### ISSUE TITLE
can not use nn::Functional(torch::softmax(-1)) in Sequential

### ISSUE THREAD

**Issue by fatalfeel** [pytorch/pytorch#41955]
nn::Sequential(nn::Linear(dim_states, h_neurons),
							nn::Functional(torch::tanh),
							nn::Linear(h_neurons, h_neurons),
	                        nn::Functional(torch::tanh),
							nn::Linear(h_neurons, dim_acts),
							nn::Functional(torch::softmax(-1)));

compile error
../src/actorcritic.cpp:13:40: error: no matching function for call to ‘softmax(int)’

nn::Functional(torch::softmax((int64_t)-1)  
error: no matching function for call to ‘softmax(int64_t)’

**Comment by glaringlee** [comment]
@fatalfeel 
softmax should take a float/double tensor, a dimension, and an optional dtype as input. It doesn't support a single number.
```
Tensor softmax(const Tensor &self, Dimname dim, c10::optional<ScalarType> dtype = c10::nullopt)
Tensor softmax(const Tensor &self, int64_t dim, c10::optional<ScalarType> dtype = c10::nullopt)
```
